diff -ur anon-proxy.old/CAInfoService.cpp anon-proxy-00.02.39/CAInfoService.cpp
--- anon-proxy.old/CAInfoService.cpp	2007-05-23 18:33:54.000000000 +0000
+++ anon-proxy-00.02.39/CAInfoService.cpp	2007-05-23 18:37:49.000000000 +0000
@@ -33,6 +33,9 @@
 #include "CASocketAddrINet.hpp"
 #include "CAUtil.hpp"
 #include "xml/DOM_Output.hpp"
+
+#include <algorithm>
+
 extern CACmdLnOptions options;
 
 
@@ -157,7 +160,7 @@
 				else
 					{
 						double dTmp=(double)diffTraffic/(double)avgTraffic;
-						tmpTraffic=min(SINT32(50.*dTmp),100);
+						tmpTraffic=std::min(SINT32(50.*dTmp),100);
 					}
 				set64(m_lastMixedPackets,tmpPackets);
 			
