diff -ur blitz++.old/blitz/funcs.h blitz++-0.9/blitz/funcs.h
--- blitz++.old/blitz/funcs.h	2005-10-08 01:22:55.000000000 +0000
+++ blitz++-0.9/blitz/funcs.h	2007-12-11 23:39:56.000000000 +0000
@@ -31,6 +31,8 @@
 #include <blitz/promote.h>
 #include <blitz/prettyprint.h>
 
+#include <cstdlib>
+
 BZ_NAMESPACE(blitz)
     
 /* Helper functions */
diff -ur blitz++.old/blitz/mathfunc.h blitz++-0.9/blitz/mathfunc.h
--- blitz++.old/blitz/mathfunc.h	2005-05-18 20:22:38.000000000 +0000
+++ blitz++-0.9/blitz/mathfunc.h	2007-12-11 23:39:39.000000000 +0000
@@ -12,6 +12,8 @@
  #include <blitz/prettyprint.h>
 #endif
 
+#include <cstdlib>
+
 BZ_NAMESPACE(blitz)
 
 // abs(P_numtype1)    Absolute value
