diff -ur bmconf.old/src/parttypes.c bmconf-0.8.4/src/parttypes.c
--- bmconf.old/src/parttypes.c	2000-09-16 07:50:11.000000000 +0000
+++ bmconf-0.8.4/src/parttypes.c	2007-05-07 17:55:56.000000000 +0000
@@ -52,20 +52,18 @@
 char * typetostring(unsigned char type)
 {
     static char retval[30];
-    char * str;
+    const char * str;
 
     void masol(char *dest, char *source)
     {
 	memcpy(dest,source,strlen(source)+1);
     }
-    
-    masol(retval,partition_type[type]);
-    if (!retval)
-    {
-	asprintf(&str,"%x",type);
-        masol(retval,str);
-	free(str);
-    }
+   
+	str=partition_type[type];
+	if (str)
+    	masol(retval,partition_type[type]);
+	else
+		snprintf(retval,sizeof(retval),"%x",type);
     return retval;
 }
 
