diff -ur cxxtools.old/include/cxxtools/arg.h cxxtools-1.4.3.6/include/cxxtools/arg.h
--- cxxtools.old/include/cxxtools/arg.h	2007-04-19 18:51:03.000000000 +0000
+++ cxxtools-1.4.3.6/include/cxxtools/arg.h	2007-12-09 20:37:11.000000000 +0000
@@ -23,6 +23,7 @@
 #define CXXTOOLS_ARG_H
 
 #include <sstream>
+#include <string.h>
 
 namespace cxxtools
 {
diff -ur cxxtools.old/src/net.cpp cxxtools-1.4.3.6/src/net.cpp
--- cxxtools.old/src/net.cpp	2007-04-29 09:38:38.000000000 +0000
+++ cxxtools-1.4.3.6/src/net.cpp	2007-12-09 20:32:31.000000000 +0000
@@ -20,6 +20,7 @@
  */
 
 #include "cxxtools/net.h"
+#include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <errno.h>
diff -ur cxxtools.old/src/syserror.cpp cxxtools-1.4.3.6/src/syserror.cpp
--- cxxtools.old/src/syserror.cpp	2007-01-24 20:14:32.000000000 +0000
+++ cxxtools-1.4.3.6/src/syserror.cpp	2007-12-09 20:33:38.000000000 +0000
@@ -20,6 +20,7 @@
  */
 
 #include "cxxtools/syserror.h"
+#include <string.h>
 #include <errno.h>
 #include <sstream>
 
diff -ur cxxtools.old/src/tcpstream.cpp cxxtools-1.4.3.6/src/tcpstream.cpp
--- cxxtools.old/src/tcpstream.cpp	2007-04-29 09:38:56.000000000 +0000
+++ cxxtools-1.4.3.6/src/tcpstream.cpp	2007-12-09 20:34:44.000000000 +0000
@@ -20,6 +20,7 @@
  */
 
 #include "cxxtools/tcpstream.h"
+#include <string.h>
 #include <sys/poll.h>
 #include <unistd.h>
 #include <netdb.h>
diff -ur cxxtools.old/src/udp.cpp cxxtools-1.4.3.6/src/udp.cpp
--- cxxtools.old/src/udp.cpp	2007-04-29 09:39:11.000000000 +0000
+++ cxxtools-1.4.3.6/src/udp.cpp	2007-12-09 20:35:44.000000000 +0000
@@ -22,6 +22,7 @@
 #include "cxxtools/udp.h"
 #include "cxxtools/dynbuffer.h"
 #include "cxxtools/log.h"
+#include <string.h>
 #include <netdb.h>
 #include <sys/poll.h>
 #include <errno.h>
