diff -ur gdb.old/gdb/remote.c gdb-6.7.1/gdb/remote.c
--- gdb.old/gdb/remote.c	2007-10-05 21:36:56.000000000 +0000
+++ gdb-6.7.1/gdb/remote.c	2007-12-11 21:48:57.000000000 +0000
@@ -1683,7 +1683,7 @@
 {
   struct remote_state *rs = get_remote_state ();
   char *limit;
-  int count, resultcount, done;
+  int count, resultcount, done = 0;
 
   resultcount = 0;
   /* Assume the 'q' and 'M chars have been stripped.  */
diff -ur gdb.old/gdb/symtab.c gdb-6.7.1/gdb/symtab.c
--- gdb.old/gdb/symtab.c	2007-12-11 21:44:03.000000000 +0000
+++ gdb-6.7.1/gdb/symtab.c	2007-12-11 21:58:47.000000000 +0000
@@ -2249,7 +2249,7 @@
 struct symtab *
 find_line_symtab (struct symtab *symtab, int line, int *index, int *exact_match)
 {
-  int exact;
+  int exact = 0;
 
   /* BEST_INDEX and BEST_LINETABLE identify the smallest linenumber > LINE
      so far seen.  */
