diff -ur hoichess.old/src/book.cc hoichess-0.10.2/src/book.cc
--- hoichess.old/src/book.cc	2007-10-30 16:12:52.000000000 +0000
+++ hoichess-0.10.2/src/book.cc	2007-12-21 19:33:16.000000000 +0000
@@ -27,6 +27,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <string.h>
 
 #include <algorithm>
 #include <fstream>
diff -ur hoichess.old/src/chess/move.cc hoichess-0.10.2/src/chess/move.cc
--- hoichess.old/src/chess/move.cc	2007-10-30 16:12:52.000000000 +0000
+++ hoichess-0.10.2/src/chess/move.cc	2007-12-21 19:35:18.000000000 +0000
@@ -29,6 +29,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 /* TODO Inline them.
  * Problem: Board::is_{valid,legal}_move() are not yet declared in
diff -ur hoichess.old/src/evalcache.cc hoichess-0.10.2/src/evalcache.cc
--- hoichess.old/src/evalcache.cc	2007-10-30 16:12:52.000000000 +0000
+++ hoichess-0.10.2/src/evalcache.cc	2007-12-21 19:34:18.000000000 +0000
@@ -24,6 +24,7 @@
 #include "common.h"
 #include "evalcache.h"
 
+#include <limits.h>
 
 EvaluationCache::EvaluationCache(unsigned long size)
 {
diff -ur hoichess.old/src/historytable.h hoichess-0.10.2/src/historytable.h
--- hoichess.old/src/historytable.h	2007-10-30 16:12:52.000000000 +0000
+++ hoichess-0.10.2/src/historytable.h	2007-12-21 19:33:56.000000000 +0000
@@ -25,6 +25,8 @@
 
 #include "basic.h"
 
+#include <string.h>
+
 class HistoryTable
 {
       private:
diff -ur hoichess.old/src/pgn.cc hoichess-0.10.2/src/pgn.cc
--- hoichess.old/src/pgn.cc	2007-10-30 16:12:52.000000000 +0000
+++ hoichess-0.10.2/src/pgn.cc	2007-12-21 19:34:45.000000000 +0000
@@ -28,6 +28,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <string.h>
 
 #include <sstream>
 
