diff -ur ipe.old/src/ipelib/ipedct.cpp ipe-6.0pre30/src/ipelib/ipedct.cpp
--- ipe.old/src/ipelib/ipedct.cpp	2006-12-23 15:05:08.000000000 +0000
+++ ipe-6.0pre30/src/ipelib/ipedct.cpp	2007-12-30 17:44:03.000000000 +0000
@@ -32,6 +32,8 @@
 
 #include "ipebase.h"
 
+#include <cstring>
+
 // --------------------------------------------------------------------
 
 // DCT component info
diff -ur ipe.old/src/ipelib/ipepdfwriter.cpp ipe-6.0pre30/src/ipelib/ipepdfwriter.cpp
--- ipe.old/src/ipelib/ipepdfwriter.cpp	2006-12-23 15:05:08.000000000 +0000
+++ ipe-6.0pre30/src/ipelib/ipepdfwriter.cpp	2007-12-30 17:45:06.000000000 +0000
@@ -39,6 +39,8 @@
 #include "ipepdfwriter.h"
 #include "ipefontpool.h"
 
+#include <cstring>
+
 typedef std::vector<IpeBitmap>::const_iterator BmIter;
 
 // --------------------------------------------------------------------
