diff -ur log4cxx.old/src/socketoutputstream.cpp log4cxx-0.9.7/src/socketoutputstream.cpp
--- log4cxx.old/src/socketoutputstream.cpp	2004-05-10 12:38:06.000000000 +0000
+++ log4cxx-0.9.7/src/socketoutputstream.cpp	2007-12-21 18:17:06.000000000 +0000
@@ -17,6 +17,7 @@
 #include <log4cxx/helpers/socketoutputstream.h>
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/loglog.h>
+#include <string.h>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
diff -ur log4cxx.old/src/stringtokenizer.cpp log4cxx-0.9.7/src/stringtokenizer.cpp
--- log4cxx.old/src/stringtokenizer.cpp	2004-05-10 12:38:08.000000000 +0000
+++ log4cxx-0.9.7/src/stringtokenizer.cpp	2007-12-21 18:21:50.000000000 +0000
@@ -15,6 +15,7 @@
  */
  
 #include <log4cxx/helpers/stringtokenizer.h>
+#include <string.h>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
diff -ur log4cxx.old/src/timezone.cpp log4cxx-0.9.7/src/timezone.cpp
--- log4cxx.old/src/timezone.cpp	2004-05-10 12:38:13.000000000 +0000
+++ log4cxx-0.9.7/src/timezone.cpp	2007-12-21 18:24:32.000000000 +0000
@@ -16,6 +16,7 @@
  
 #include <log4cxx/helpers/timezone.h>
 #include <locale>
+#include <string.h>
 
 
 int getYear(int64_t date)
diff -ur log4cxx.old/src/xmlsocketappender.cpp log4cxx-0.9.7/src/xmlsocketappender.cpp
--- log4cxx.old/src/xmlsocketappender.cpp	2004-05-10 12:38:16.000000000 +0000
+++ log4cxx-0.9.7/src/xmlsocketappender.cpp	2007-12-21 18:51:30.000000000 +0000
@@ -22,6 +22,7 @@
 #include <log4cxx/xml/xmllayout.h>
 #include <log4cxx/level.h>
 #include <log4cxx/helpers/transform.h>
+#include <string.h>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
