diff -ur pump.old/pump.c pump-0.8.24/pump.c
--- pump.old/pump.c	2007-12-10 02:30:16.000000000 +0000
+++ pump-0.8.24/pump.c	2007-12-10 03:03:10.000000000 +0000
@@ -652,7 +652,7 @@
 		    if (intf[i].set & PUMP_NETINFO_HAS_HOSTNAME)
 			strncpy(cmd.u.status.hostname,
 			    intf->hostname, sizeof(cmd.u.status.hostname));
-		    cmd.u.status.hostname[sizeof(cmd.u.status.hostname)] = '\0';
+		    cmd.u.status.hostname[sizeof(cmd.u.status.hostname) - 1] = '\0';
 
 		    if (intf[i].set & PUMP_NETINFO_HAS_DOMAIN)
 			strncpy(cmd.u.status.domain,
