--- searchandrescue.old/sar/string.cpp	2007-06-19 22:59:19.000000000 +0000
+++ searchandrescue-0.8.2/sar/string.cpp	2007-06-19 23:02:14.000000000 +0000
@@ -29,7 +29,7 @@
 #ifdef __MSW__
 int strcasecmp(const char *s1, const char *s2);
 #endif
-char *strcasestr(const char *haystack, const char *needle);
+char *strcasestr(const char *haystack, const char *needle) throw();
 int strpfx(const char *s, const char *pfx);
 int strcasepfx(const char *s, const char *pfx);
 
@@ -218,7 +218,7 @@
  *	Case insensitive version of strstr(). Returns the pointer to
  *	needle in haystack if found or NULL on no match.
  */
-char *strcasestr(const char *haystack, const char *needle)
+char *strcasestr(const char *haystack, const char *needle) throw()
 {
 	const char *strptr1, *strptr2, *strptr3;
 
