diff -ur speech-tools.old/config/compilers/gcc_defaults.mak speech-tools-1.2.3/config/compilers/gcc_defaults.mak
--- speech-tools.old/config/compilers/gcc_defaults.mak	2007-06-22 20:52:10.000000000 +0000
+++ speech-tools-1.2.3/config/compilers/gcc_defaults.mak	2007-06-22 20:54:01.000000000 +0000
@@ -74,12 +74,12 @@
 PROFILE_gprof_CXXFLAGS  = -pg
 PROFILE_gprof_LINKFLAGS = -pg
 
-SHARED_CCFLAGS  = -fPIC -fno-shared-data
-SHARED_CXXFLAGS  = -fPIC -fno-shared-data
-SHARED_LINKFLAGS = -fno-shared-data
+SHARED_CCFLAGS  = -fPIC
+SHARED_CXXFLAGS  = -fPIC
+SHARED_LINKFLAGS =
 
 ifndef GCC_MAKE_SHARED_LIB
-    MAKE_SHARED_LIB = gcc -shared -fno-shared-data -o XXX
+    MAKE_SHARED_LIB = gcc -shared -o XXX
 else
     MAKE_SHARED_LIB = $(GCC_MAKE_SHARED_LIB)
 endif
