--- vertex-0.1.15/vertex/string.cpp	2007-06-10 06:37:30.000000000 +0000
+++ vertex.old/vertex/string.cpp	2007-06-10 06:38:13.000000000 +0000
@@ -30,7 +30,7 @@
 #ifdef __MSW__
 int strcasecmp(const char *s1, const char *s2);
 #endif
-char *strcasestr(const char *haystack, const char *needle) throw();
+char *strcasestr(const char *haystack, const char *needle);
 int strpfx(const char *s, const char *pfx);
 int strcasepfx(const char *s, const char *pfx);
 
@@ -219,7 +219,7 @@
  *	Case insensitive version of strstr(). Returns the pointer to
  *	needle in haystack if found or NULL on no match.
  */
-char *strcasestr(const char *haystack, const char *needle) throw()
+char *strcasestr(const char *haystack, const char *needle)
 {
 	const char *strptr1, *strptr2, *strptr3;
 
