diff -ur xalan.old/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp xalan-1.10/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp
--- xalan.old/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2005-05-05 00:32:04.000000000 +0000
+++ xalan-1.10/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2007-12-29 18:36:08.000000000 +0000
@@ -16,8 +16,6 @@
 #if !defined(DIRECTORY_ENUMERATOR_HEADER_GUARD_1357924680)
 #define DIRECTORY_ENUMERATOR_HEADER_GUARD_1357924680
 
-
-
 // Base header file.  Must be first.
 #include <xalanc/PlatformSupport/PlatformSupportDefinitions.hpp>
 
@@ -36,6 +34,7 @@
 
 #include <functional>
 #include <iterator>
+#include <cstring>
 
 
 #include "xercesc/framework/MemoryManager.hpp"
diff -ur xalan.old/c/src/xalanc/XalanDOM/XalanDOMString.cpp xalan-1.10/c/src/xalanc/XalanDOM/XalanDOMString.cpp
--- xalan.old/c/src/xalanc/XalanDOM/XalanDOMString.cpp	2005-08-08 15:25:52.000000000 +0000
+++ xalan-1.10/c/src/xalanc/XalanDOM/XalanDOMString.cpp	2007-12-29 18:25:29.000000000 +0000
@@ -19,6 +19,7 @@
 
 
 #include <cassert>
+#include <cstring>
 
 
 
diff -ur xalan.old/c/src/xalanc/XalanExe/XalanExe.cpp xalan-1.10/c/src/xalanc/XalanExe/XalanExe.cpp
--- xalan.old/c/src/xalanc/XalanExe/XalanExe.cpp	2005-04-29 21:39:46.000000000 +0000
+++ xalan-1.10/c/src/xalanc/XalanExe/XalanExe.cpp	2007-12-29 18:41:19.000000000 +0000
@@ -19,6 +19,7 @@
 
 
 #include <cstdlib>
+#include <cstring>
 #if defined(XALAN_CLASSIC_IOSTREAMS)
 #include <iostream.h>
 #else
diff -ur xalan.old/c/src/xalanc/XMLSupport/FormatterToHTML.cpp xalan-1.10/c/src/xalanc/XMLSupport/FormatterToHTML.cpp
--- xalan.old/c/src/xalanc/XMLSupport/FormatterToHTML.cpp	2005-06-30 21:52:35.000000000 +0000
+++ xalan-1.10/c/src/xalanc/XMLSupport/FormatterToHTML.cpp	2007-12-29 18:26:53.000000000 +0000
@@ -26,6 +26,7 @@
 
 
 #include <cassert>
+#include <cstring>
 
 
 
diff -ur xalan.old/c/src/xalanc/XSLT/ElemNumber.cpp xalan-1.10/c/src/xalanc/XSLT/ElemNumber.cpp
--- xalan.old/c/src/xalanc/XSLT/ElemNumber.cpp	2005-07-26 20:09:17.000000000 +0000
+++ xalan-1.10/c/src/xalanc/XSLT/ElemNumber.cpp	2007-12-29 18:30:55.000000000 +0000
@@ -15,7 +15,7 @@
  */
 #include "ElemNumber.hpp"
 
-
+#include <cstring>
 
 #include <xercesc/sax/AttributeList.hpp>
 
